src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -308,6 +386,9 @@ private static class Log
             private static readonly Action<ILogger, Exception> _handshakeFailed =
                 LoggerMessage.Define(LogLevel.Error, new EventId(5, "HandshakeFailed"), "Failed connection handshake.");
 
+            private static readonly Action<ILogger, Exception> _failedWritingMessage =
+                LoggerMessage.Define(LogLevel.Trace, new EventId(6, "FailedWritingMessage"), "Failed writing message.");
```

Comment:
Include the message type in the log message? Also the level should probably be higher than LogLevel.Trace.

Reply:
I think it needs to be trace. Nobody should care about write failures. We saw lots of feedback from kestrel and dialed this down. It's something that will happen all the time

Reply:
I think this is `Debug`-worthy, but I don't have the context from Kestrel. My general rules have been:

* `Information` - Always useful to every user
* `Debug` - Sometimes useful to a human user investigating an issue
* `Trace` - Only ever useful to automation

This message seems like it would be useful to a human user to diagnose why messages aren't arriving on a client (it's an exceptional case, we just don't consider it a full-on error).